python/ray/data/_internal/logical/rules/set_read_parallelism.py

```diff
@@ -43,18 +43,18 @@ def compute_additional_split_factor(
 
     available_cpu_slots = ray_available_resources().get("CPU", 1)
     if (
-        parallelism
+        parallelism > 0
```

Comment:
Nit: might be easier to understand as "user-specified parallelism"
```suggestion
        parallelism != -1
```